sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/xml/StaxXmlParser.scala

```diff
@@ -280,20 +267,45 @@ class StaxXmlParser(
     while (!shouldStop) {
       parser.nextEvent match {
         case e: StartElement =>
-          kvPairs +=
-            (UTF8String.fromString(StaxXmlParserUtils.getName(e.asStartElement.getName, options)) ->
-            convertField(parser, valueType))
+          val key = UTF8String.fromString(
+            StaxXmlParserUtils.getName(e.asStartElement.getName, options))
+          try {
+            kvPairs += key -> convertField(parser, valueType)
+          } catch {
+            case partialValueException: PartialValueException =>
+              badRecordException = badRecordException.orElse(Some(partialValueException.cause))
+              StaxXmlParserUtils.skipChildren(parser)
+              kvPairs += key -> partialValueException.partialResult
+            case NonFatal(e) =>
+              badRecordException = badRecordException.orElse(Some(e))
+              StaxXmlParserUtils.skipChildren(parser)
+          }
         case c: Characters if !c.isWhiteSpace =>
           // Create a value tag field for it
-          kvPairs +=
+          val key = UTF8String.fromString(options.valueTag)
           // TODO: We don't support an array value tags in map yet.
-          (UTF8String.fromString(options.valueTag) -> convertTo(c.getData, valueType))
+          try {
+            kvPairs += (key -> convertTo(c.getData, valueType))
+          } catch {
+            case partialValueException: PartialValueException =>
+              badRecordException = badRecordException.orElse(Some(partialValueException.cause))
+              StaxXmlParserUtils.skipChildren(parser)
+              kvPairs += key -> partialValueException.partialResult
+            case NonFatal(e) =>
+              badRecordException = badRecordException.orElse(Some(e))
+              StaxXmlParserUtils.skipChildren(parser)
```

Comment:
Will this skip other valid <key, value> pairs?

Reply:
It will not. This is the exception branch.